service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/search/builder/FACatchSearchBuilder.java

```diff
@@ -119,16 +117,46 @@ protected void appendSelectGroupColumns(List<GroupCriteria> groupByFieldList, St
      * @param sql
      */
     protected void enrichWherePartOFQueryForDISOrDIM(StringBuilder sql) {
-        sql.append(" and ( a.typeCode ='").append(FishingActivityTypeEnum.FISHING_OPERATION.toString()).append("' and faCatch.typeCode IN ('").append(FaCatchTypeEnum.DEMINIMIS).append("','").append(FaCatchTypeEnum.DISCARDED).append("')) ");
+        // This is an ugly workaround to handle the fact that we can't be 100% sure there are previous search criteria
+        if (!sql.toString().trim().toUpperCase().endsWith("WHERE")) {
+            sql.append(" AND ");
+        }
+
+        sql.append("( a.typeCode ='")
+                .append(FishingActivityTypeEnum.FISHING_OPERATION.toString())
+                .append("' AND faCatch.typeCode IN ('")
+                .append(FaCatchTypeEnum.DEMINIMIS)
+                .append("','")
+                .append(FaCatchTypeEnum.DISCARDED)
+                .append("')) ");
     }
 
     protected void conditionsForFACatchSummaryReport(StringBuilder sql) {
-        sql.append(" and (" +
-                "(a.typeCode ='").append(FishingActivityTypeEnum.FISHING_OPERATION.toString()).append("' and faCatch.typeCode IN('").append(FaCatchTypeEnum.ONBOARD)
-                .append("','").append(FaCatchTypeEnum.KEPT_IN_NET).append("','").append(FaCatchTypeEnum.BY_CATCH).append("'))  OR (a.typeCode ='").append(FishingActivityTypeEnum.RELOCATION.toString())
-                .append("' and a.relatedFishingActivity.typeCode='").append(FishingActivityTypeEnum.JOINED_FISHING_OPERATION.toString()).append("' and a.vesselTransportGuid = a.relatedFishingActivity.vesselTransportGuid  ")
-                .append(" and faCatch.typeCode IN('").append(FaCatchTypeEnum.ONBOARD).append("','").append(FaCatchTypeEnum.TAKEN_ON_BOARD)
-                .append("','").append(FaCatchTypeEnum.ALLOCATED_TO_QUOTA).append("')))");
+        // This is an ugly workaround to handle the fact that we can't be 100% sure there are previous search criteria
```

Comment:
It would probably be a good thing to replace all this sql string handling with javax.persistence.criteria.CriteriaBuilder, but I don't know if it is worth the effort. At least we have to make sure we don't open up for sql injection.

Reply:
Yes, it is not pretty...